zookeeper-server/src/main/java/org/apache/zookeeper/client/ConnectStringParser.java

```diff
@@ -70,16 +69,22 @@ public ConnectStringParser(String connectString) {
         List<String> hostsList = split(connectString, ",");
         for (String host : hostsList) {
             int port = DEFAULT_PORT;
-            try {
-                String[] hostAndPort = ConfigUtils.getHostAndPort(host);
+            String[] hostAndPort = NetUtils.getIPV6HostAndPort(host);
+            if (hostAndPort.length != 0) {
                 host = hostAndPort[0];
                 if (hostAndPort.length == 2) {
                     port = Integer.parseInt(hostAndPort[1]);
                 }
-            } catch (ConfigException e) {
-                e.printStackTrace();
+            } else {
```

Comment:
Why this part of the logic is not inside 
NetUtils.getIPV6HostAndPort(host) ?
What about pushing it up there?

Reply:
I wanted to make getIPV6HostAndPort  API more generic so it can be used other places as well for example it can be used to input of  host:port:port and [host]:port:port  as well. If we move above said logic to getIPV6HostAndPort  host:port:port will give wrong result; So I wanted to keep only common logic in  getIPV6HostAndPort.

Reply:
makes sense to me